docs/usage.md

```diff
@@ -0,0 +1,309 @@
+# Usage
+- [Selecting Java distribution](#Selecting-Java-distribution)
+  - [Supported version syntax](#Supported-version-syntax)
+  - [Zulu](#Zulu)
+  - [Adoptium](#Adoptium)
+- [Installing custom Java package type](#Installing-custom-Java-package-type)
+- [Installing custom Java architecture](#Installing-custom-Java-architecture)
+- [Installing custom Java distribution from local file](#Installing-Java-from-local-file)
+- [Testing against different Java versions](#Testing-against-different-Java-versions)
+- [Testing against different Java distributions](#Testing-against-different-Java-distributions)
+- [Publishing using Apache Maven](#Publishing-using-Apache-Maven)
+- [Publishing using Gradle](#Publishing-using-Gradle)
+
+See [action.yml](../action.yml) for more details on task inputs.
+
+## Selecting Java distribution
+Input `distribution` is mandatory and should be provided to use action. See [Supported distributions](../README.md#Supported-distributions) for the list of available options.
+
+```yaml
+steps:
+- uses: actions/checkout@v2
+- uses: actions/setup-java@v2-preview
+  with:
+    distribution: '<distribution>' # See 'Supported distributions' for available distribution
+    java-version: '11.x'
+- run: java -cp java HelloWorldApp
+```
```

Comment:
this section almost fully duplicates Basic section from the main README, but we announced it as Advanced usage. 
I'd suggest removing it from here and moving Adoptium and Zulu examples here instead (even above the "Supported version syntax" section) because the main header is "Selecting Java distribution". 
And section "Supported version syntax" I'd make a 2-level header, because it doesn't really relate to distributors.